### DISCUSSION TITLE
Need help with install: Pillow error

### DISCUSSION THREAD

**Discussion by MoisesHernandezSanchez** [mathoudebine/turing-smart-screen-python#344] [Q&A] [answered]
Greetings! When I try running the "python.exe -m pip install -r requirements.txt" command in VSC, I get the following error:
File "<string>", line 1005, in <module>
      __main__.RequiredDependencyException:
     
      The headers or library files could not be found for zlib,
      a required dependency when compiling Pillow from source.

ERROR: Failed building wheel for Pillow
Failed to build Pillow
ERROR: Could not build wheels for Pillow, which is required to install pyproject.toml-based projects

Any advice? Running on Win10, using VSC and hoping to use it for the XuanFang 3.5"I bought of AliE, since its official software got flagged as adware by Malwarebytes.

**Comment by mathoudebine** [comment]
Hello @MoisesHernandezSanchez! What is your Python version? You can find it by running `python.exe --version`
Did you download it from the official Python website or from the Microsoft Store?

**Reply by MoisesHernandezSanchez** [reply]
Hi thanks for the really fast reply!
Ran the command, got this output:
Python 3.8.2
odd since I installed the recent python 12 a few days ago.

**Reply by mathoudebine** [reply]
Maybe your `python.exe` does not link to the latest Python available on your system. What is the output of the command `py.exe --version`?

**Reply by MoisesHernandezSanchez** [reply]
Seems like that might be the case, running the command py.exe --version outputs 
Python 3.12.0
while python.exe --version gives
Python 3.8.2

Any idea how I can fix that?

**Reply by mathoudebine** [reply]
You can try to uninstall the old Python 3.8.2 from the Control Panel or Settings.
After uninstalling, test again the `python.exe --version` to check if it has been reset the the newer Python 3.12.
If not, you may need to reinstall Python 3.12 and during the setup don't forget to check "Add Python 3.12 to PATH"
![0 7nOyowsPsGI19pZT](https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/1e257df7-cd1a-449f-95bb-662327a98378)

**Reply by MoisesHernandezSanchez** [reply]
Alright, no longer have that issue, now have this one:
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/33675521/4ac77bf2-3fcf-4813-a4e8-3f99acfe7ab4)

But we're making progress it seems!

**Reply by mathoudebine** [reply]
Do you have a virtual environment created in VSC? If so you have to delete it because it is linked to your old Python installation.
Does the error appear when you do `python.exe -m pip install -r requirements.txt`?
Can you try to run this command from a command-line outside VSC?

**Reply by mathoudebine** [reply, marked as answer]
@MoisesHernandezSanchez after investigating, Python 3.12 just went out 2 days ago and PIL Pillow library we use in this project does not yet support Python 3.12: https://pillow.readthedocs.io/en/stable/installation.html

For now you need to keep Python 3.11 until this issue is fixed. Don't worry, Python 3.11 is still actively supported until 2027-10!
You can uninstall Python 3.12 and install Python 3.11 from here: https://www.python.org/downloads/release/python-3116/ downloads are at the end of the page

**Reply by MoisesHernandezSanchez** [reply]
Python 3.11.6 works like a charm, and the program is now functional! thank you for being awesome!

**Reply by mathoudebine** [reply]
Thanks and have fun!
